### PR TITLE
Feature/aapd 672

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-672

## Description of change

- Adds a v2 endpoint to docs api that talks to back office storage account
- Couldn't get the sas url or user delegation to work locally

## Checklist

- [x] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: docker-compose.yml
Comment (bryn-pins) [inline docker-compose.yml:1]: some commented out bits regarding using a self signed cert for the azurite container in here, couldn't seem to get fully working though

File: packages/appeals-service-api/package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
 	"author": "Planning-Inspectorate <https://gov.uk/pins>",
 	"dependencies": {
 		"@azure/app-configuration": "^1.3.1",
-		"@azure/identity": "3.0.1",
+		"@azure/identity": "^4.0.1",
```
Comment (bryn-pins) [inline packages/appeals-service-api/package.json:34]: the breaking change is just requiring node 18

File: packages/common/src/client/blob-storage-client.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+	/**
+	 * uploads file to container, requires create permissions for the given container
+	 * @param {string} containerName - Blob storage container name
+	 * @param {string} blobName - Blob name, may include path
+	 * @param {Object<string,string>} metadata - PINs Data Model metadata, requires contentType
+	 * @param {import('node:stream').Readable} fileStream - file
+	 * @returns {Promise<import('@azure/storage-blob').BlobUploadCommonResponse>}
+	 */
+	async uploadBlob(containerName, blobName, metadata, fileStream) {
```
Comment (bryn-pins) [inline packages/common/src/client/blob-storage-client.js:161]: uploadBlob is currently unused so could potentially remove this

File: packages/document-service-api/src/lib/blobStorage.js [hunk 3]
```diff
@@ -73,6 +73,9 @@ const getMetadataForAllFiles = async (containerClient, applicationId) => {
 
 const getMetadataForSingleFile = async (containerClient, applicationId, documentId) => {
 	try {
+		// todo: retrieve single file metadata only, below example needs testing
+		// return await containerClient.getBlockBlobClient(`${applicationId}/${documentId}`).getProperties();
```
Comment (Si3rr4wow) [inline packages/document-service-api/src/lib/blobStorage.js:77]: 🙌

**Review by Si3rr4wow — DISMISSED** [review]
v nice

**Review by Si3rr4wow — APPROVED** [review]

**Pull request merged**